script.js

```diff
@@ -0,0 +1,157 @@
+
+const userEntry = document.querySelector('#display');
+const btnContainer = document.querySelector('#button-container');
+const operBtns = document.querySelectorAll('.oper');
+const backspace = document.querySelector('#backspace');
+
+//handles keyboard entries
+window.addEventListener('keydown', event => {
+    if (!isNaN(event.key)) userEntry.textContent += event.key;
+
+    if (oper.hasOwnProperty(event.key)) {
+        userEntry.textContent += event.key;
+    };
+
+    if (event.key === '/') userEntry.textContent += '÷';
+
+    if (event.key === 'Enter') {
+        const result = shuntingYard(userEntry.textContent);
+        userEntry.textContent = result;
+    };
+});
+
+//handles button clicks
+btnContainer.addEventListener('click', button => {
+    if (button.target.id !== 'button-container') {
+        if (button.target.id !== 'equals' && button.target.parentNode.id !== 'edit') {
+            userEntry.textContent += button.target.textContent;
+        };
+
+        if (button.target.classList.contains('oper')) {
+            operBtns.forEach(btn => btn.style.color = 'white');
+            button.target.style.color = 'orange';
+        } else {
+            operBtns.forEach(btn => btn.style.color = 'white');
+        };
+    };
+    if (button.target.id === 'equals') {
+        const result = shuntingYard(userEntry.textContent);
+        userEntry.textContent = result;
+    };
+    if (button.target.id === 'cancel') userEntry.textContent = '';
+
+    if (button.target.id === 'backspace') userEntry.textContent = userEntry.textContent.slice(0, -1);
+});
+
+
+function add(numOne, numTwo) {
+    //prevents numbers being concatenated 
+    const a = Number(numOne);
+    const b = Number(numTwo);
+    const result = (a + b).toString()
+    return result;
+};
+
+function subtract(numOne, numTwo) {
+    return numOne - numTwo;
+};
+
+function multiply(numOne, numTwo) {
+    return numOne * numTwo;
+};
+
+function divide(numOne, numTwo) {
+    return numOne / numTwo;
+};
+
+function percentOf(numOne, numTwo) {
+    //divide by 100 to get 1% of numTwo
+    return (numTwo / 100) * numOne;
+}
+
+
+const oper = {
+    '+': {
+        fn: add,
+        prec: 1,
+    },
+    '-': {
+        fn: subtract,
+        prec: 1,
+    },
+    '*': {
+        fn: multiply,
+        prec: 2,
+    },
+    '÷': {
+        fn: divide,
+        prec: 2,
+    },
+};
+
+
+function shuntingYard(infixExpr) {
+    const expr = infixExpr.split('');
+    const output = [];
+    const stack = [];
+    let accum = '';
+    for (let i = 0; i < expr.length; i++) {
+        
+        if (isNaN(expr[i]) && expr[i] !== '.') {
+            if (expr[i] === '(' || !stack.length || stack[stack.length - 1] === '(') {
+                stack.push(expr[i]);
+            } else if (expr[i] !== ')') {
+                while (stack[stack.length - 1] !== '(' && stack.length && oper[expr[i]].prec <= oper[stack[stack.length - 1]].prec) {
+                    output.push(stack.pop());
```

Comment:
split it out onto multiple lines and store the value for the next index , its horrible to read: 

oper[stack[stack.length-1].prec

becomes 

const top =stack.length-1
const next= stack[top]
oper[next]